cfme/common/vm.py

```diff
@@ -991,18 +1011,20 @@ def _select_rows(indexes):
 
 
 @attr.s
-class VMCollection(BaseVMCollection):
+class VMCollection(BaseVMCollection, RetirementMixin):
     ENTITY = VM
 
-    def retire(self, entities):
+    def retire(self, **kwargs):
         """Select and then retire one or more VMs|Instances from the All VMs|Instances page.
 
         Args:
-            entities (list): List of one or more InfraVM | Instance objects. This method
+            entities: :py:class:`list` List of one or more VM instances This method
                 is intended to be called from either an InfraVMCollection instance (with a list
-                of InfraVM objects, or an InstanceCollection instance (with a list of Instance
+                of InfraVM objects), or an InstanceCollection instance (with a list of Instance
                 objects).
         """
+        entities = kwargs.get('entities', None)
+        assert entities, "retire requires a list of VMs."
```

Comment:
Rather than `assert` I think this should just raise a `ValueError` or something similar.

Reply:
Changed.